### PR TITLE
allergies: Use enumeration for allergens

### LINKED ISSUE #605: allergies: Use enum for allergens
The current `allergies` exercise represents allergens as strings. As they are a fixed set of values, we should be using an enum instead.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closed #605

### CODE REVIEW

File: generators/Exercises/Generators/Allergies.cs [hunk 1]
```diff
@@ -11,19 +14,34 @@ protected override void UpdateTestMethod(TestMethod testMethod)
...
+
+        private UnescapedValue[] ConvertExpected(dynamic expected)
+        {
+            if (expected is string[] allergens)
+                return allergens.Select(RenderAllergen).ToArray();
+
+            return Array.Empty<UnescapedValue>();
+        }
+
+        private UnescapedValue RenderAllergen(dynamic input) => Render.Enum("Allergen", input);
```
Comment (robkeim) [inline generators/Exercises/Generators/Allergies.cs]: Minor: does `nameof(Allergen)` work here?

**Comment by robkeim** [comment]
Thanks @ErikSchierboom 👍

File: generators/Exercises/Generators/Allergies.cs (hunk 1 above)
Comment (ErikSchierboom) [reply generators/Exercises/Generators/Allergies.cs]: Nope, because the `Allergen` enumeration is not defined in the generators project :)

**Comment by ErikSchierboom** [comment]
Thanks for merging!

File: generators/Exercises/Generators/Allergies.cs (hunk 1 above)
Comment (robkeim) [reply generators/Exercises/Generators/Allergies.cs]: *face palm* yup makes sense :)

**Pull request merged**